### PR TITLE
Maintenance mode

### PR DESCRIPTION
**Pull request by khenderick**

### CODE REVIEW

**Comment by khenderick** [comment]
@itolegmysin @iOlivier these changes introduce maintenance mode on the portal. For now, it's only on the local gateway if the correct feature is announced by the gateway. By changing the `index.js` navigation we'll be able to also use this code to add maintenance mode on the cloud once it's implemented there as well.

File: src/components/websocket.js [hunk 1]
```diff
@@ -39,6 +39,7 @@ export class WebSocketClient {
         this.name = socketEndpoint;
         this.lastDataReceived = 0;
         this.reconnectFrequency = 1000 * 60;
+        this.binary = binary === undefined ? true : !!binary;
```
Comment (itolegmysin) [inline src/components/websocket.js]: It looks a little strange, may you explain this ?
So if `binary` is `null` or `''` then `this.binary` is `false`, else `this.binary` is `true` ?

File: src/pages/settings/maintenance.html [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+                <div class="box box-warning">
+                    <div class="box-header with-border">
+                        <h3 class="box-title">
+                            <span style="position: relative; top: 1px; padding-right: 5px;">
+                                ${'pages.settings.maintenance.title' & t}
+                            </span>
+                        </h3>
+                    </div>
+                    <div class="box-body">
+                        <div id="terminal" style="background-color: black; padding: 10px; max-height: 500px; overflow: auto;"></div>
```
Comment (itolegmysin) [inline src/pages/settings/maintenance.html]: I'd like to make all styles in a separate file (styles.css) 😃

File: src/components/websocket.js (hunk 1 above)
Comment (khenderick) [reply src/components/websocket.js]: Yes, binary mode is active by default, so `undefined` means we set it to true. All other possible values are loosely evaluated

**Pull request merged**